x-pack/plugins/viz_editor/common/components/index_pattern_panel/index_pattern_panel.tsx

```diff
@@ -4,39 +4,97 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiButtonEmpty, EuiIcon, ICON_TYPES } from '@elastic/eui';
+import {
+  EuiIcon,
+  // @ts-ignore
+  EuiSuperSelect,
+  ICON_TYPES,
+} from '@elastic/eui';
 // @ts-ignore untyped dependency
 import { palettes } from '@elastic/eui/lib/services';
-import React, { useState } from 'react';
+import zipObject from 'lodash-es/zipObject';
+import React, { useEffect, useState } from 'react';
+import chrome from 'ui/chrome';
```

Comment:
It's not that important at the moment, but these kind of imports (stateful stuff from other plugins) will be gone on the new platform and have to be loaded in plugin lifecycle hooks (https://github.com/elastic/kibana/pull/31147/files?short_path=f1b3221#diff-f1b322112aece16fe3b028a685be85c1). This part of the new platform isn't ready yet, but we can prepare and only include stateful imports from other modules (`ui/...`, `plugins/...` and `uiExports/...`) in or near our entrypoint. I tried to do this for the pipeline-specific stuff - the imports are all in `x-pack/plugins/viz_editor/public/app.tsx` and passed down via props or parameters